src/main.rs

```diff
@@ -23,11 +23,23 @@ mod platforms;
 mod softeners;
 mod system_io;
 
-use crate::comm::dbus::{control_interface::ControlInterface, status_interface::StatusInterface};
+use crate::{
+    comm::dbus::{control_interface::ControlInterface, status_interface::StatusInterface},
+    platforms::platform::register_platform,
+    softeners::xilinx_dfx_mgr::XilinxDfxMgrPlatform,
+};
+
+fn register_platforms() {
+    // TODO: implement a proc_macro that registers the platform with compat string
+    register_platform("xlnx,zynqmp-pcap-fpga,versal-fpga,zynq-devcfg-1.0", || {
+        Box::new(XilinxDfxMgrPlatform::new())
+    });
+}
```

Comment:
```suggestion
fn register_platforms() {
    // TODO: implement a proc_macro that registers the platform with compat string
    register_platform("xlnx,zynqmp-pcap-fpga,versal-fpga,zynq-devcfg-1.0", || {
        Box::new(XilinxDfxMgrPlatform::new())
    });
     register_platform("universal", || {
        Box::new(UniversalPlatform::new())
    });
}
```
I don't think `get_platform_for_known_platform` defaults here so we need to register universal as well so that the user can say "universal" if they want to bypass dfx-mgr for example

Reply:
Oh yeah I didnt thought of this usecase, will update in next commit